packages/facade/src/apis/sheets/f-sheet-hooks.ts

```diff
@@ -14,20 +14,37 @@
  * limitations under the License.
  */
 
-import type { IDisposable, Nullable } from '@univerjs/core';
-import { Inject, toDisposable } from '@univerjs/core';
+import type { IDisposable, IExecutionOptions, ISheetDataValidationRule, Nullable } from '@univerjs/core';
+import { ICommandService, Inject, Injector, toDisposable } from '@univerjs/core';
+import type { IRuleChange, IValidStatusChange } from '@univerjs/data-validation';
+import { DataValidationModel } from '@univerjs/data-validation';
+import type { IUpdateCommandParams } from '@univerjs/docs';
+import type { IAddSheetDataValidationCommandParams, IRemoveSheetAllDataValidationCommandParams, IRemoveSheetDataValidationCommandParams, IUpdateSheetDataValidationOptionsCommandParams, IUpdateSheetDataValidationRangeCommandParams, IUpdateSheetDataValidationSettingCommandParams } from '@univerjs/sheets-data-validation';
+import { AddSheetDataValidationCommand, RemoveSheetAllDataValidationCommand, RemoveSheetDataValidationCommand, UpdateSheetDataValidationOptionsCommand, UpdateSheetDataValidationRangeCommand, UpdateSheetDataValidationSettingCommand } from '@univerjs/sheets-data-validation';
 
 import type { IDragCellPosition, IHoverCellPosition } from '@univerjs/sheets-ui';
 import { DragManagerService, HoverManagerService } from '@univerjs/sheets-ui';
+import type { CommentUpdate, IAddCommentCommandParams, IDeleteCommentCommandParams } from '@univerjs/thread-comment';
+import { AddCommentCommand, DeleteCommentCommand, DeleteCommentTreeCommand, ThreadCommentModel, UpdateCommentCommand } from '@univerjs/thread-comment';
 
 export class FSheetHooks {
```

Comment:
Whether to put it in FSheetHooks still needs discussion and consensus.   discussion in https://github.com/dream-num/univer-pro/issues/2244

Reply:
immigrate to workbook now